### ISSUE TITLE
`getDefaultProps` overrides properties passed from parent component

### ISSUE THREAD

**Issue by asdf404** [yahoo/react-intl#107]
Hi. There is a problem with passing of `messages` to children of my top component. Using of `getDefaultProps` overrides any messages passed from parent. I've using it for default translations, maybe there is another way to do it.

[Here](https://github.com/asdf404/react-intl-bug) I've prepare an example of bug.

If I [comment](https://github.com/asdf404/react-intl-bug/blob/921c8a840811b3b03de013b13f09d5b5db24df8a/index.js#L11) `getDefaultProps` in my `ChildComponent` then test passes.

**Comment by ericf** [comment]
What behavior were you expecting? A merge of `messages`?

You could do this:

```js
const ChildComponent = React.createClass({
    mixins: [ReactIntl.IntlMixin],

    getDefaultProps() {
        return {
            messages: Object.assign({}, this.context.message, {
                foo: 'Foo'
            })
        }
    },

    render() {
        return (
            <FormattedMessage message={this.getIntlMessage('foo')} />
        );
    }
});
```

**Comment by asdf404** [comment]
Thanks for answer. I've expected override of default messages with new.

Your example doesn't work. I've got an error: `TypeError: Cannot read property 'messages' of undefined`.

Looks like `this.context` and `this.props` is undefined in `getDefaultProps`.

Before I've used a hack like this:

```js
render() {
    this.props.messages = this.props.messages ? this.props.messages : getDefaultMessages();
    // ...
}
```

But now I've removed it and pass `messages` property explicitly.
